LEGO1/tgl/d3drm/group.cpp

```diff
@@ -89,7 +89,10 @@ Result GroupImpl::Add(const Group* pGroup)
 // FUNCTION: LEGO1 0x100a3430
 Result GroupImpl::Add(const MeshBuilder* pMeshBuilder)
 {
-	const MeshBuilderImpl* pMeshBuilderImpl = static_cast<const MeshBuilderImpl*>(pMeshBuilder);
+	const MeshBuilderImpl* pMeshBuilderImpl = dynamic_cast<const MeshBuilderImpl*>(pMeshBuilder);
+	if (!pMeshBuilderImpl) {
+		return Result::Error;
```

Comment:
Log a message?

Reply:
We could add a small template helper similar to UnalignedRead which does the cast and has a logging (assert?) on failure. I don't think conversion should ever fail

Reply:
I can add the logging but could you do the cleanups as a follow up?

Reply:
That's ok! The extra proposed template function is out-of-scope for this pr.